Add package startup message, update LICENSE

TylerSagendorf opened the pull request:
- Fixes #1 
- Display message in the console with the versions of all pericircle packages when library(pericircle) is first called in the current R session. 
- Update LICENSE to include the tidyverse MIT license, and include proper attribution in the README.
- @bcjaeger I updated the DESCRIPTION to make you the package creator and maintainer, since you came up with the idea for this package.

<img width="507" height="71" alt="pericircle package startup message." src="https://github.com/user-attachments/assets/b271190a-99f0-480d-9277-f9597c22be9a" />

TylerSagendorf committed:
Add package startup message, update LICENSE

Display message in the console with the versions of all pericircle packages when library(pericircle) is first called in the current R session. Update LICENSE to include the tidyverse MIT license, and include proper attribution in the README.

bcjaeger commented on DESCRIPTION:
```diff
@@ -1,9 +1,12 @@
+Version: 0.0.0.9001
 Authors@R: c(
-    person("Tyler", "Sagendorf", , "tsagendorf@perisphere-rwe.com", 
+    person(given = "Byron", family = "Jaeger", 
+           email = "bcjaeger@perisphere-rwe.com", 
```
I think you should be first author.

TylerSagendorf committed:
Add packages to Imports

bcjaeger commented on R/attach.R:
```diff
@@ -25,3 +27,53 @@ pericircle_attach <- function() {
+  }
+
+  header <- cli::rule(
+    left = cli::style_bold("Attaching core pericircle packages"),
```
we can use `@importFrom` if you like that approach for handling dependencies. Feel free to use `pkg::function` though if you like that better.

TylerSagendorf replied:
I had just copied this directly from `tidyverse` and added `@import` lines to R/pericircle-package.R. I can change this to add `@importFrom` lines to the necessary functions, if we want to be consistent with, e.g., `perinary`.

bcjaeger reviewed (commented):
This looks fantastic, thank you Tyler! I left some minor comments for you to consider. I also tested on my laptop and I noticed the load message did not include `pericircumference`. Any thoughts on why this occurs on my system but not yours?

<img width="911" height="212" alt="image" src="https://github.com/user-attachments/assets/f21db3df-8285-4ae4-a777-1d693508f39c" />

TylerSagendorf commented:
> This looks fantastic, thank you Tyler! I left some minor comments for you to consider. I also tested on my laptop and I noticed the load message did not include `pericircumference`. Any thoughts on why this occurs on my system but not yours?
> 
> <img alt="image" width="911" height="212" src="https://private-user-images.githubusercontent.com/11821781/512261747-f21db3df-8285-4ae4-a777-1d693508f39c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjI3OTE2MjksIm5iZiI6MTc2Mjc5MTMyOSwicGF0aCI6Ii8xMTgyMTc4MS81MTIyNjE3NDctZjIxZGIzZGYtODI4NS00YWU0LWE3NzctMWQ2OTM1MDhmMzljLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTEwVDE2MTUyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU5Y2MxMzAyNGQ5OWUxYjg1NjI0ODg1ZDE4MDgyNGNjZWE5ZTZhNGFlNjhmZDMwZjVlNDBhZDg3ZmIyM2FiMzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BfeMhgfHQMgtMxGQQX9MogNHiBzHIF1ntS9bzslxcx4">

That will happen if you already have one of the packages attached. Can you try `library(pericircle)` in a fresh R session?

TylerSagendorf committed:
Prefer importFrom over import tag

Import each function, rather than using the :: syntax, to be consistent with other Perisphere R packages.

TylerSagendorf committed:
Update package authorship

bcjaeger commented:
Ah ha, I did start from a fresh R session but I forgot my Rprofile is this. Looks like everything is working as intended. =]

<img width="607" height="257" alt="image" src="https://github.com/user-attachments/assets/6e9460e4-52cf-42dc-8fd1-a1e5c168d338" />

bcjaeger commented:
🥳
